src/env.cpp

```diff
@@ -418,7 +418,7 @@ NAN_METHOD(EnvWrap::batchWrite) {
     Nan::Callback* callback;
     Local<Value> options = info[1];
 
-    if (!info[1]->IsNull() && !info[1]->IsUndefined() && info[1]->IsObject()) {
+    if (!info[1]->IsNull() && !info[1]->IsUndefined() && info[1]->IsObject() && !info[1]->IsFunction()) {
```

Comment:
Since this one also fixes issues introduced by your PR, please squash this one too.